File: LEETCODE/Problems/(Easy) 118. Pascal's Triangle.py
from collections import defaultdict


class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        pascal = defaultdict(lambda: defaultdict(lambda: 0))
        pascal[1][1] = 1
        for i in range(2, numRows+1):
            for j in range(1, i+1):
                pascal[i][j] = pascal[i-1][j-1]+pascal[i-1][j]

        res = []
        for i in range(1, numRows+1):
            temp = []
            for j in range(1, i+1):
                temp.append(pascal[i][j])
            res.append(temp)
        return res

File: LEETCODE/Problems/(Easy) 118. Pascal's Triangle.java
import java.util.ArrayList;
import java.util.List;
class Solution
{
    public List<List<Integer>> generate(int numRows) 
    {
        int [][] pascal = new int[numRows][numRows];
        for (int i = 0; i < numRows; i++)
        {
            for (int j = 0; j <= i;j++)
            {
                if (j==0||j==i)
                {
                    pascal[i][j]=1;
                }
                else
                {
                    pascal[i][j]=pascal[i-1][j-1]+pascal[i-1][j];
                }
            }
        }
        List<List<Integer>> res= new ArrayList();
        for (int i = 0; i < numRows; i++)
        {
            List<Integer> temp= new ArrayList();
            for (int j = 0; j <= i; j++)
            {
                temp.add(pascal[i][j]);
            }
            res.add(temp);
        }
        return res;
    }
}